File: selenium-automation-java/src/main/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage {

    private WebDriver driver;
    private By usernameField = By.id("userName");
    private By passwordField = By.id("password");
    private By loginButton = By.id("login");
    private By logoutButton = By.id("submit");

    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void goToLoginPage() {
        driver.get("https://demoqa.com/login");
    }

    public void login(String username, String password) {
        driver.findElement(usernameField).sendKeys(username);
        driver.findElement(passwordField).sendKeys(password);
        driver.findElement(loginButton).click();
    }

    public void logout() {
        driver.findElement(logoutButton).click();
    }
}

File: playwright-tests-ts/src/pages/login-page.ts
import { Page } from '@playwright/test';

/**
 * LoginPage - Page Object Model for login page
 * Encapsulates selectors and common actions related to login
 */
export class LoginPage {
  readonly page: Page;
  readonly usernameField = '#userName';
  readonly passwordField = '#password';
  readonly loginButton = '#login';
  readonly logoutButton = '#submit';

  constructor(page: Page) {
    this.page = page;
  }

  async goToLoginPage(): Promise<void> {
    await this.page.goto('https://demoqa.com/login');
  }

  async login(username: string, password: string): Promise<void> {
    await this.page.fill(this.usernameField, username);
    await this.page.fill(this.passwordField, password);
    await this.page.click(this.loginButton);
  }

  async logout(): Promise<void> {
    await this.page.click(this.logoutButton);
  }

  async getErrorMessage(): Promise<string | null> {
    return this.page.textContent('#name');
  }
}

